### PR TITLE
Change javelin bag in More Survival Tools to work like quivers

### PR DESCRIPTION
**Pull request by zachary-kaelan**
#### Summary
SUMMARY: Bugfixes "Fixed javelin bag and allowed them to hold fishing spears."

#### Purpose of change
Fixes #30128 - Fix javelin bag in More Survival Tools to work with javelin ammo, which was already fixed, but then it was changed back in #30338.

A stack of ammo counts as a single item, meaning a stack of javelins just goes into the first holster slot. 

Fishing spears have similar enough volume and stats, and don't have any storage options of their own, so I took out two birds with one stone.

#### Describe the solution
Simply switched it back to acting like the quiver.

#### Describe alternatives you've considered
The reason why it was changed back in the first place was so that you could store multiple javelin types. Having a holster allowed for that. But it was *extremely* counter intuitive.

It says it holds 10 items, so a player is naturally going to create 10 javelins to store in the bag. But if you have 10 javelins in a stack, the bag says "you have no suitable weapons available", when you activate it. I just thought it was broken until I did some digging and realized that I had to throw a few javelins out of my inventory to split the stack, because the stack had an accumulative volume that was above the limit. I'm assuming if you had too few javelins, it would be below the volume limit, and not work there either. There is no way for the player to know any of this.

If one wants to store multiple types of javelins, because weaker javelins have higher range, they can simply wear multiple javelin bags. If the encumbrance is an issue, then add javelin straps worn on the legs.

#### Additional context
Javelins and fishing spears are similar enough that making the two interchangeable in the future, or setting them further apart, may be beneficial.

### CODE REVIEW

**Comment by Highmongrel** [comment]
When were javelins changed to stack in hand? When I made the change to holster back they didn't, and they don't in base game.

**Comment by EirenexTheDragon** [comment]
> 
> 
> When were javelins changed to stack in hand? When I made the change to holster back then they didn't, and they don't in base game.

More Survival Tools would change how javelins work alltogether because the atlatl uses javelins as ammo. that was an early issue because the javelin bag i made only was for vanilla and wouldnt work properly with MST since that turns javelins into ammo for quivers and the such. if you dont use mods, javelins wouldnt stack like ammo does.

**Comment by Highmongrel** [comment]
> 
> More Survival Tools would change how javelins work alltogether because the atlatl uses javelins as ammo. that was an early issue because the javelin bag i made only was for vanilla and wouldnt work properly with MST since that turns javelins into ammo for quivers and the such. if you dont use mods, javelins wouldnt stack like ammo does.

Yes, but back then they didn't stack in hand even with MST active. I just tested it on an older build. When they didn't stack in hand the holster function worked as intended with 1 javelin going into each slot.

File: data/json/items/armor.json [hunk 1]
```diff
@@ -9571,15 +9570,12 @@
...
-      "max_volume": "1250 ml",
-      "draw_cost": 30,
-      "flags": [ "JAVELIN" ]
+      "type": "bandolier",
+      "capacity": 5,
+      "ammo": [ "javelin", "fishspear" ],
+      "draw_cost": 30
     },
-    "flags": [ "WATER_FRIENDLY", "OVERSIZE", "BELTED" ]
+    "flags": [ "WATER_FRIENDLY", "OVERSIZE" ]
```
Comment (Rail-Runner) [inline data/json/items/armor.json]: Is this intended? Some kind of bag which covers torso would most likely be the same layer as a backpack (therefore the BELTED flag).

**Comment by Highmongrel** [comment]
You had to put them in one at a time, which was unfortunate I admit. In retrospect leaving at quiver was probably a better idea.

The stacking in hand thing does kinda break the item in a purely hand throwing sense in that it removes the "reload" time though. Not really part of this PR though.

**Comment by zachary-kaelan** [comment]
Hey guys, it won't compile. And to rub it in, when I first tried the last commit I made, my entire computer blue screened.

**Comment by Highmongrel** [comment]
data/mods/More_Survival_Tools/armor.json needs to be linted

**Comment by zachary-kaelan** [comment]
Didn't know formatting was an issue, but finally got it to work. Ready to merge, boys.

**Pull request closed without merging**